### DISCUSSION TITLE
Upload large video> The video is taking too much time to load and for big files the download button is disabled in mobile app

### DISCUSSION THREAD

**Discussion by sachinb80** [Sunbird-Ed/Community#522] [General Discussion]
We are using the 4.6 version of Sunbird 
The uploaded video is taking too much time to load
**What do you think we could do to reduce the load time and enable the download button for big files?**?
The Videos take time to load on KEF App
Over Wifi

1. 500 mb = 1 min 55 sec
2. 1 gb = 3 min 53 sec
3. 100 mb = 27 seconds

Over Mobile network(5g)

1. 500 mb = 4 min 5 sec
2. 1 gb = 5 min 30 sec
3. 100 mb = 30 sec

This is my observation
Also, the download button is disabled for 500Mb and 1GB files but it is enabled for 100MB files

**This file is 100MB**
![Screenshot_2023-06-06-16-51-23-492_org key app-edit](https://github.com/Sunbird-Ed/Community/assets/109504832/5cfdf14c-77ed-45e5-8731-798745db20a7)
**This file is 500MB**
![Screenshot_2023-06-06-16-51-39-662_org key app-edit](https://github.com/Sunbird-Ed/Community/assets/109504832/53b07573-3cf1-44f8-8a83-6728f0dcebc1)
[@vinukumar-vs](https://github.com/vinukumar-vs) [@rajeevsathish](https://github.com/rajeevsathish) @[rajeshkumaravel](https://github.com/rajeshkumaravel) [@Krishnaj20](https://github.com/Krishnaj20)

**Comment by swayangjit** [comment]
@sachinb80  The download button is disabled if there is no downloadUrl in metaData. There is a limit config in Knowledge Service if the size exceeds a specific limit they will delete the downloadUrl in content metadata and regarding loading time it depends on the Network provider.

**Reply by sachinb80** [reply]
@swayangjit Thanks for the reply.
